### PR TITLE
Add learnable positional channels

### PR DESCRIPTION
**Pull request by jder**
## Summary
- allow configuring extra learned positional channels
- incorporate positional parameters into Samudra model forward pass
- add unit test verifying positional parameters update via gradients

## Testing
- `uvx pre-commit run --all-files`
- `uv run pytest -m "not manual and not cuda"` *(fails: FileNotFoundError for remote dataset)*
- `uv run pytest tests/test_positional_channels.py -q`

------
https://chatgpt.com/codex/tasks/task_e_689dd13368e4832085aed233390cc422

### CODE REVIEW

**Comment by jder** [comment]
Ok, this seems to be [learning something](https://openathena.slack.com/archives/C08CYM42DT3/p1755892720158529?thread_ts=1755888945.806759&cid=C08CYM42DT3) and [improves over the new baseline in some places](https://wandb.ai/m2lines/ocean-emulators/workspace?nw=6q3j4ecg44n) so I think we can keep it around for a bit and perhaps do another experiment or two here.

**Comment by alxmrs** [comment]
Maybe this is what you wanted to talk about today: should I give this a review? Are the other similar PRs worthy of review?

**Comment by jder** [comment]
@alxmrs yes, this is ready for review

File: tests/test_positional_channels.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        TensorMap.init_instance("thetao_1", "hfds")
+        coords = {"lev": [0], "y": np.arange(h), "x": np.arange(w)}
+        zero = np.zeros((1, h, w))
+        data = xr.Dataset(
+            {
+                "thetao": (("lev", "y", "x"), zero),
+                "hfds": (("y", "x"), np.zeros((h, w))),
+            },
+            coords=coords,
+        )
```
Comment (alxmrs) [inline tests/test_positional_channels.py]: Make it a little more consistent with the `ones` example below.

```suggestion
        data = xr.Dataset(
            {
                "thetao": (("lev", "y", "x"), np.zeros((1, h, w))),
                "hfds": (("y", "x"), np.zeros((h, w))),
            },
            coords=coords,
        )
```

File: tests/test_positional_channels.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            torch.ones(h, w),
+            torch.ones(h, w),
+        )
+
+        config = SamudraConfig(
+            ch_width=[2, 2],
+            n_out=1,
+            dilation=[1],
+            n_layers=[1],
+            pos_channels=1,
```
Comment (alxmrs) [inline tests/test_positional_channels.py:48]: A few comments would be helpful additions to this integration test. For example, that this is the "Act" step.

File: tests/test_positional_channels.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            static_data=None,
+        )
+
+        assert model.positional_params is not None
+        assert model.positional_params.shape == (config.pos_channels, h, w)
+        assert not torch.allclose(
+            model.positional_params.detach(),
+            torch.zeros_like(model.positional_params),
+        )
+
```
Comment (alxmrs) [inline tests/test_positional_channels.py:65]: Another comment here to describe what this section of the test would be helpful.

File: tests/test_positional_channels.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        )
+        model = Samudra(
+            config=config,
+            hist=0,
+            wet=torch.ones(1, h, w, dtype=torch.bool),
+            area_weights=torch.ones(h, w),
+            static_data=None,
+        )
+
+        assert model.positional_params is not None
```
Comment (alxmrs) [inline tests/test_positional_channels.py:59]: 🐑  request: a comment for this block, or strings in the asserts.

File: src/ocean_emulators/config.py [hunk 5]
```diff
@@ -217,6 +217,7 @@ class SamudraConfig(BaseConfig):
     last_kernel_size: int = 3
     pad: str = "circular"
     wet: Any | None = None
+    pos_channels: int = 0
```
Comment (alxmrs) [inline src/ocean_emulators/config.py]: 🐑 Maybe a comment here to help understand what "pos" means, or else a more descriptive variable name.

**Review by alxmrs — APPROVED** [review]
All minor suggestions, this LGTM.
